client/zecale/dummy_app/__main__.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+#
+# SPDX-License-Identifier: LGPL-3.0+
+
+
+from zecale.dummy_app.deploy import deploy
+from zecale.dummy_app.get import get
+from zeth.cli.constants import ETH_NETWORK_FILE_DEFAULT
+from zeth.cli.utils import get_eth_network
+from click import group, option, pass_context, Context
+from click_default_group import DefaultGroup  # type: ignore
+from typing import Optional
+
+
+@group(cls=DefaultGroup, default_if_no_args=True, default="--help")
+@option(
+    "--eth-network",
+    help="Ethereum RPC endpoint, network or config file "
+    f"(default: '{ETH_NETWORK_FILE_DEFAULT}')")
```

Comment:
(sometimes both are done, as in the commented lines in https://github.com/clearmatics/zecale/pull/42#discussion_r504006470 for instance)

Reply:
TBH I have never considered this specifically.  It's seems to just be a side-effect of trying to avoid providing constants unless they are required, and trying to make the command arg types match any function parameters, and keep unnecessary logic out of the command body.

For example, we have a util function for the network which accepts an `Optional[str]`, so it's redundant to give a `default` in every command that needs this param (unfortunately we do still need to include it for the help text - I can't see a nice way around that)

For the `--dispatcher-instance-file`, there is no util function that falls back to the default.  Using the param `default` avoids an `Optional` and logic in the command to handle it.

Reply:
I see. Thanks